src/ui/popup/manage-settings-page/import-settings.tsx

```diff
@@ -0,0 +1,64 @@
+import {m} from 'malevic';
+import {ViewProps} from '../types';
+import ControlGroup from '../control-group';
+import {UserSettings} from '../../../definitions';
+import {Button} from '../../controls';
+import {openFile} from '../../utils';
+import {DEFAULT_SETTINGS} from '../../../defaults';
+
+export default function ImportButton(props: ViewProps) {
+
+    function getValidatedObject<T>(source: any, compare: T): Partial<T>  {
+        const result = {};
+        if (source == null || (typeof source !== 'object' && !Array.isArray(source))) { 
```

Comment:
Use `source == null || typeof source !== 'object' || Array.isArray(source)` so that only plain objects pass.